src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,73 @@
-import React from 'react';
+import React, { useState } from 'react';
+import cn from 'classnames';
 import './FindMovie.scss';
+import { getMovie } from '../../api';
+import { MovieData } from '../../types/MovieData';
+import { ResponseError } from '../../types/ReponseError';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
+
+// eslint-disable-next-line max-len
+const DEFAULT_IMAGE = 'https://via.placeholder.com/360x270.png?text=no%20preview';
+
+type Props = {
+  onMovieAdd: (movie: Movie) => void,
+};
+
+export const FindMovie: React.FC<Props> = ({ onMovieAdd }) => {
+  const [query, setQuery] = useState('');
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [foundMovie, setFoundMovie]
+    = useState<Movie | null>(null);
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setIsLoading(true);
+
+    getMovie(query)
+      .then((data: MovieData | ResponseError) => {
+        if ('Error' in data) {
```

Comment:
The error handling in the 'getMovie' promise is not ideal. If the promise rejects, the 'Error' property won't necessarily be in the data. Consider using a '.catch()' block to handle errors more effectively.